### PR TITLE
RUMM-1326 track Activity Intent action and data URI

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Track the Intent's action and data URI, only if the "track extra" option is enabled.

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/ActivityViewTrackingStrategy.kt [hunk 1]
```diff
@@ -19,7 +19,8 @@ import com.datadog.android.rum.model.ViewEvent
  * A [ViewTrackingStrategy] that will track [Activity] as RUM Views.
  *
  * Each activity's lifecycle will be monitored to start and stop RUM Views when relevant.
- * @param trackExtras whether to track Activity Intent extras
+ * @param trackExtras whether to track the Activity's Intent information (extra attributes,
+ * action, data URI)
```
Comment (0xnm) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/ActivityViewTrackingStrategy.kt:23]:
```suggestion
 * action, data URI).
```

**Review by 0xnm — APPROVED** [review]
lgtm! Added minor spelling suggestion.

File: dd-sdk-android/apiSurface [hunk 2]
```diff
@@ -1253,6 +1253,7 @@ abstract class com.datadog.android.rum.tracking.ActivityLifecycleTrackingStrateg
   override fun onActivityStopped(android.app.Activity)
   override fun onActivityCreated(android.app.Activity, android.os.Bundle?)
   override fun onActivityResumed(android.app.Activity)
+  protected fun convertToRumAttributes(android.content.Intent?): Map<String, Any?>
```
Comment (mariusc83) [inline dd-sdk-android/apiSurface:1256]: Shouldn't we also add a monitor and a nightly test for this use case ?

**Review by mariusc83 — CHANGES_REQUESTED** [review]

File: dd-sdk-android/apiSurface (hunk 2 above)
Comment (xgouchet) [reply dd-sdk-android/apiSurface:1256]: No, it'll already be taken care of by the callers (tracking activities etc…). I'm not even sure why it is public.

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**